DTS-STN/canadian-dental-care-plan
# Replacing dynatrace service with dependency injectable service

### Description
Adding `.server` modules such as service, repository, and mapper to allow `DynatraceService` to be injectable. 

Unlike the previous implementation of `dynatrace-service.server` (which has been removed in this PR), repository and mapper have been modified to throw an `Error` whenever an issue occurs. The service implementation is responsible for handling these errors by returning `null` to ensure that the application does not crash due to unhandled exceptions.

### Screenshots (if applicable)
![image](https://github.com/user-attachments/assets/647e49e5-5c99-4f37-9193-f3dd555fee50)

### Checklist

### Test Instructions
Configure `DYNATRACE_API_RUM_SCRIPT_URI` and `DYNATRACE_API_RUM_SCRIPT_TOKEN` environment variables and run the app. The RUM `<script>` should be rendered. Remove these environment variables and the `<script>` should not be rendered.
